File: Polymorfisme/Polymorfisme.py
class Makhluk:
    def desc(self):
        print("\n=== Makhluk ===")
        print("Dapat bernafas")


class Mamalia(Makhluk):
    def desc(self):
        print("\n=== Mamalia ===")
        print("Menyusui")


class Sapi(Mamalia):
    def desc(self):
        print("\n=== Sapi ===")
        print("Berkaki 4 , Hitam-Putih")


class Bertelur(Makhluk):
    def desc(self):
        print("\n=== Bertelur ===")
        print("Bertelur")


class Ayam(Bertelur):
    def desc(self):
        print("\n=== Ayam ===")
        print("Berkaki 2 , Merah")


hewan = Makhluk()
mamalia = Mamalia()
sapi = Sapi()
bertelur = Bertelur()
ayamKampung = Ayam()


hewan.desc()
mamalia.desc()
sapi.desc()
bertelur.desc()
ayamKampung.desc()

File: Polymorfisme/Polymorfisme.cpp
#include <iostream>
#include <conio.h>
using namespace std;

class Makhluk
{
public:
    void desc()
    {
        cout << "==== Makhluk =====" << endl;
        cout << "Dapat Bernafas" << endl;
    }
};

class Mamalia : public Makhluk
{
public:
    void desc()
    {
        cout << "=== Mamalia ===" << endl;
        cout << "Menyusui" << endl;
    }
};

class Bertelur : public Makhluk
{
public:
    void desc()
    {
        cout << "=== Bertelur ==== " << endl;
        cout << "Bertelur" << endl;
    }
};

class Sapi : public Mamalia
{
public:
    void desc()
    {
        cout << "=== Sapi ===" << endl;
        cout << "Berkaki 4, Hitam-putih" << endl;
    }
};

class Ayam : public Bertelur
{
public:
    void desc()
    {
        cout << "=== Ayam ===" << endl;
        cout << "Berkaki 2, Merah" << endl;
    }
};

int main()
{
    //deklarasi objek
    Makhluk hewan;
    Mamalia mamalia;
    Bertelur bertelur;
    Sapi sapi;
    Ayam ayamKampung;

    hewan.desc();
    cout << endl;

    hewan = mamalia;
    mamalia.desc();

    hewan = sapi;
    sapi.desc();
    cout << endl;

    hewan = bertelur;
    bertelur.desc();

    hewan = ayamKampung;
    ayamKampung.desc();

    return 0;
}